accounts/abi/bind/bind.go

```diff
@@ -139,118 +139,166 @@ var bindType = map[Lang]func(kind abi.Type) string{
 	LangJava: bindTypeJava,
 }
 
+// Helper function for the binding generators.
+// It reads the unmatched characters after the inner type-match,
+//  (since the inner type is a prefix of the total type declaration),
+//  looks for valid arrays (possibly a dynamic one) wrapping the inner type,
+//  and returns the sizes of these arrays.
+//
+// Returned array sizes are in the same order as solidity signatures; inner array size first.
+// Array sizes may also be "", indicating a dynamic array.
+func wrapArray(stringKind string, innerLen int, innerMapping string) (string, []string) {
```

Comment:
It would be simpler, more readable and arguably more efficient to use the following regular expression `\[(\d*)\]` along with `FindAllStringSubmatch`.

Reply:
Yup, probably a good idea. I'll simplify the implementation 👍